### ISSUE TITLE
ArcGIS WebAdaptor 11.1 always reconfigured

### ISSUE THREAD

**Issue by Biboba** [Esri/arcgis-powershell-dsc#466]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 4.1.0

### Affected Resource(s)

* ArcGIS_WebAdaptor

### Configuration Files

[GisPortalAGSDeployment_anonymous.txt](https://github.com/Esri/arcgis-powershell-dsc/files/11412325/GisPortalAGSDeployment_anonymous.txt)

### Expected Behavior

If webAdaptor is already configured, it should not be configured again.

### Actual Behavior

If webAdaptor is already configured, it is configured again.

### Steps to Reproduce

Run the attached configuration several times: even if web adaptor is already configured, it is configured again.
Here is the log with a webadaptor already installed:

```
05.05.2023 08:30:02: [AGS01]: LCM:  [ Start  Resource ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptorAGS01]
05.05.2023 08:30:02: [AGS01]: LCM:  [ Start  Test     ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptorAGS01]
05.05.2023 08:30:02: [AGS01]: LCM:  [ End    Test     ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptorAGS01]  in 0.0680 seconds.
05.05.2023 08:30:02: [AGS01]: LCM:  [ Start  Set      ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptorAGS01]
05.05.2023 08:30:02: [AGS01]:                            [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptorAGS01] https://AGS01.company.com/arcgis/webadaptor
05.05.2023 08:30:07: [AGS01]:                            [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptorAGS01] Output of execution:- 
Configuring the ArcGIS Enterprise Web Adaptor with ArcGIS Server.

Configuring Web adaptor...

The ArcGIS Web Adaptor has been successfully configured.
05.05.2023 08:30:07: [AGS01]: LCM:  [ End    Set      ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptorAGS01]  in 2.6780 seconds.
05.05.2023 08:30:07: [AGS01]: LCM:  [ End    Resource ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptorAGS01]

```

### Important Factoids

I debug a little bit and I think the culprit is here:
https://github.com/Esri/arcgis-powershell-dsc/blob/8621f7a7256d6f8b9431caf4674e01c8d3bce249/Modules/ArcGIS/DSCResources/ArcGIS_WebAdaptor/ArcGIS_WebAdaptor.psm1#L241

$WAConfig object is properly retrieved but $WAConfig .url does not exist. It should be $WAConfig.**gisserver**.url or $WAConfig.**portal**.url depending on the component to register.

### References

Thanks !

**Comment by Biboba** [comment]
Hi @cameronkroeker,
Any feedback on this issue ? Were you able to reproduce ?
Thanks

**Comment by cameronkroeker** [comment]
Hi @Biboba,

Great catch! I was able to reproduce it. Here is a workaround that worked for me:

**Change:**
https://github.com/Esri/arcgis-powershell-dsc/blob/8621f7a7256d6f8b9431caf4674e01c8d3bce249/Modules/ArcGIS/DSCResources/ArcGIS_WebAdaptor/ArcGIS_WebAdaptor.psm1#L241

**To:**
```
$WAConfigSiteUrl = if($Component -ieq "Portal"){ $WAConfig.portal.url }else{ $WAConfig.gisserver.url }
```

Thanks,
Cameron K.

**Comment by cameronkroeker** [comment]
> The workaround does not work for me:
> 
> ```
> [[ArcGIS_InstallPatch]PortalInstallPatch]
> PowerShell DSC resource ArcGIS_InstallPatch  failed to execute Set-TargetResource
> functionality with error message: Error downloading remote file. Error - Exception calling "DownloadFile" with "2"
> argument(s): "The remote server returned an error: (404) Not Found."
> At C:\Program Files\WindowsPowerShell\Modules\ArcGIS\4.1.0\ArcGIS.psm1:261 char:5
> +     Trace-DSCJob -Job $Job -JobName $ConfigurationName -DebugMode $De ...
> +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
>     + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Trace-DSCJob
> ```
> 
> ![image](https://user-images.githubusercontent.com/112546295/252720280-11af4f39-8998-4da1-b6b3-b125f99b604c.png)
> 
> Any ideas? @Biboba @cameronkroeker

Hi @widsuoids,

Looks like a different issue. It's failing to download portal patches from the internet. The suggested workaround for this topic is to solve an issue where the Webadaptor always configures with portal/server even though it has been configured on previous attempts.

Thanks,
Cameron K.

**Comment by cameronkroeker** [comment]
This issue has been addressed in release v4.20.

https://github.com/Esri/arcgis-powershell-dsc/releases/tag/v4.2.0
